File: php/features.php
<?php
declare(strict_types=1);

// AutomaticWeatherStations SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class AutomaticWeatherStationsFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new AutomaticWeatherStationsBaseFeature();
            case "test":
                return new AutomaticWeatherStationsTestFeature();
            default:
                return new AutomaticWeatherStationsBaseFeature();
        }
    }
}

File: py/features.py
# AutomaticWeatherStations SDK feature factory

from feature.base_feature import AutomaticWeatherStationsBaseFeature
from feature.test_feature import AutomaticWeatherStationsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: AutomaticWeatherStationsBaseFeature(),
        "test": lambda: AutomaticWeatherStationsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
